sourced/ml/core/models/id_splitter.py

```diff
@@ -0,0 +1,131 @@
+    Reference: https://arxiv.org/abs/1805.11651
+    """
+    NAME = "id_splitter_bilstm"
+    VENDOR = "source{d}"
+    DESCRIPTION = "Weights of the BiLSTM network to split source code identifiers."
+    LICENSE = DEFAULT_LICENSE
+
+    DEFAULT_MAXLEN = 40
+    DEFAULT_PADDING = "post"
+    DEFAULT_MAPPING = {c: i for i, c in enumerate(string.ascii_lowercase, start=1)}
+    DEFAULT_BATCH_SIZE = 4096
+
+    def construct(self, model: "keras.models.Model",
+                  maxlen: int = DEFAULT_MAXLEN,
+                  padding: str = DEFAULT_PADDING,
+                  mapping: dict = DEFAULT_MAPPING,
+                  batch_size: int = DEFAULT_BATCH_SIZE) -> "IdentifierSplitterNN":
+        """
+        :param model: keras model used for identifier splitting.
+        :param maxlen: maximum length of input identifers.
+        :param padding: where to pad the identifiers of length < maxlen. Can be "left" or "right".
+        :param mapping: { str: int } mapping of characters to integers.
+        :param batch_size: batch size of input data fed to the model.
+        :return: BiLSTM based source code identifier splitter.
+        """
+
+        self._maxlen = maxlen
+        self._padding = padding
+        self._mapping = mapping
+        self._model = model
+        self._batch_size = batch_size
+        self._log = logging.getLogger("IdentifierSplitterBiLSTM")
+
+        return self
+
+    @property
+    def model(self) -> "keras.models.Model":
+        """
+        Returns the wrapped keras model.
+        """
+        return self._model
+
+    @property
+    def batch_size(self) -> int:
+        """
+        Returns the batch size used to run the model
+        """
+        return self._batch_size
+
+    def _generate_tree(self) -> dict:
+        return {
+            "config": self._model.get_config(),
+            "weights": self._model.get_weights(),
+            "mapping": self._mapping,
+            "maxlen": self._maxlen,
+            "padding": self._padding,
+            }
+
+    def _load_tree(self, tree: dict):
+        model = keras.models.Model.from_config(tree["config"])
+        model.set_weights(tree["weights"])
+        self.construct(model, maxlen=tree["maxlen"],
+                       padding=tree["padding"], mapping=tree["mapping"])
+
+    def _prepare_single_identifier(self, identifier: str) -> Tuple[numpy.array, str]:
+
+        # Clean identifier
+        clean_id = "".join(char for char in identifier.lower() if char in self._mapping)
+        if len(clean_id) > self._maxlen:
+            clean_id = clean_id[:self._maxlen]
+        self._log.debug("Preprocessed identifier: %s : %s" % (identifier, clean_id))
+        return numpy.array([self._mapping[c] for c in clean_id]), clean_id
+
+    def prepare_input(self, identifiers: Sequence[str]) -> Tuple[numpy.array, List[str]]:
+        """
+        Prepares input by converting a sequence of identifiers to the corresponding
```

Comment:
```suggestion
        Prepare input by converting a sequence of identifiers to the corresponding \
```

Again. Something is definitely wrong with the check.

Reply:
Yeah that should have been checked.

Reply:
I find out why. We have a lot of style mistakes from an old and not so good style check. 
So a lot of modules are just ignored. 
https://github.com/src-d/ml-core/blob/master/.flake8

It is a lot of work to fix all style issues...